doc/source/container_builder.rst

```diff
@@ -0,0 +1,282 @@
+--------------------------------
+
+Docker packages are usually available with the used distribution.
+
+.. code:: bash
+
+    $ sudo zypper in docker
+
+Make sure that the user, who is intended to build images, is a member
+of the `docker` group. Run the following command:
+
+.. code:: bash
+
+    $ sudo useradd -G docker <builduser>
+
+It is required to logout and login again to let this change
+become active.
+
+Once this is done you need to setup the Docker storage backend.
+By default Docker uses the device mapper to manage the storage for
+the containers it starts. Unfortunately, this does not work if the
+container is supposed to build images because it runs into conflicts
+with tools like :command:`kpartx` which itself maps devices using
+the device mapper.
+
+Fortunately, there is a solution for Docker which allows us to use
+Btrfs as the storage backend. The following is only required if your
+host system root filesystem is not btrfs:
+
+.. code:: bash
+
+    $ sudo qemu-img create /var/lib/docker-storage.btrfs 20g
+    $ sudo mkfs.btrfs /var/lib/docker-storage.btrfs
+    $ sudo mkdir -p /var/lib/docker
+    $ sudo mount /var/lib/docker-storage.btrfs /var/lib/docker
+
+    $ sudo vi /etc/fstab
+
+      /var/lib/docker-storage.btrfs /var/lib/docker btrfs defaults 0 0
+
+    $ sudo vi /etc/sysconfig/docker
+
+      DOCKER_OPTS="-s btrfs"
+
+Finally start the docker service:
+
+.. code:: bash
+
+    $ sudo systemctl restart docker
+
+Installing and Setting up the Build Container
+----------------------------------------------
+
+In order to build in a contained environment Docker has to start a
+privileged system container. Such a container must be imported before
+Docker can use it. The Build Container is provided to you as a
+service and build with KIWI in the project
+at https://build.opensuse.org/project/show/Virtualization:Appliances:Images
+On a regular basis the result image is pushed
+to https://hub.docker.com/r/opensuse/dice.
+
+There are two ways to import the Build Container to your local Docker system
+
+1. Download from the openSUSE Buildservice and manually import
+2. Use :command:`docker` to pull the box from Dockerhub
+
+Pull from Dockerhub
+-------------------
+
+.. code:: bash
+
+    $ docker pull opensuse/dice:latest
+
+Download from the Open BuildService
+-----------------------------------
+
+Download the .tar.bz2 file which starts with :file:`Docker-Tumbleweed`
+
+.. code:: bash
+
+    $ wget http://download.opensuse.org/repositories/Virtualization:/Appliances:/Images/images/Docker-Tumbleweed.XXXXXXX.docker.tar.xz
+
+Import the downloaded tarball to docker as follows:
+
+.. code:: bash
+
+    $ cat Docker-Tumbleweed.XXXXXXX.docker.tar.xz | docker import - opensuse/dice:latest
+
+
+Installing and Setting up Vagrant
+---------------------------------
+
+.. note:: Optional step
+
+    This step can be skipped if there are no complex provision tasks
+    of the building environment required.
```

Comment:
Depends on what is a "complex provision task". What qualifies a task to be "complex"? :))